bndtools.core/src/bndtools/central/Central.java

```diff
@@ -607,20 +601,31 @@ public static void refreshPlugins() throws Exception {
 	}
 
 	public static void refreshPlugin(Refreshable plugin) throws Exception {
-		if (plugin.refresh()) {
+		refreshPlugin(plugin, false);
+	}
+
+	public static void refreshPlugin(Refreshable plugin, boolean force) throws Exception {
+		if (plugin.refresh() || force) {
```

Comment:
I am wondering if we should use `|` here instead of `||` to be explicit about no short-circuiting.

```java
if (plugin.refresh() | force) {
```

Reply:
Unless the JLS permits the arguments to be reordered (I don't think it does?) there doesn't seem to be any harm in short-circuiting.

Reply:
It is correct... now. But when someone later looks at this code and reorders the operands, this can subtly break things. `plugin.refresh()` must always be called. To be more explicit about this, the code could be:

```java
boolean refreshed = plugin.refresh();
if (refreshed || force) {
```

Reply:
Gotcha - but in that case, wouldn't a warning comment achieve the same thing?

Reply:
Sure. I guess I am asking for something...